### PR TITLE
VIH-7537 check for linked participant availability

### PR DESCRIPTION
**Pull request by phillipjchambers**
### JIRA link (if applicable) ###

VIH-7537

### Change description ###

* check linked participant availability
* add a test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by kevin-hmcts — APPROVED** [review]

**Review by kevin-hmcts — APPROVED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts [hunk 1]
```diff
@@ -67,4 +73,21 @@ export class IndividualParticipantStatusListComponent extends WRParticipantStatu
...
+                const linkedParticipant = this.nonJudgeParticipants.find(p => p.id === lp.linked_id);
+                return (
+                    linkedParticipant &&
+                    linkedParticipant.status !== ParticipantStatus.Available &&
+                    linkedParticipant.status !== ParticipantStatus.InConsultation
+                );
+            });
+
+            return unavailableLinkedParticipants;
+        } else {
```
[the remark this answers is not in the document]
Comment (phillipjchambers) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts:87]: Good point

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts [hunk 2]
```diff
@@ -43,15 +43,21 @@ export class IndividualParticipantStatusListComponent extends WRParticipantStatu
     }
 
     getParticipantStatusCss(participant: ParticipantResponse): string {
-        if (participant.status !== ParticipantStatus.Available && participant.status !== ParticipantStatus.InConsultation) {
```
[the remark this answers is not in the document]
Comment (phillipjchambers) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts:46]: I don't think so, it's only called from within the ngFor

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts [hunk 3]
```diff
@@ -43,15 +43,21 @@ export class IndividualParticipantStatusListComponent extends WRParticipantStatu
...
+        if (
+            (participant.status !== ParticipantStatus.Available && participant.status !== ParticipantStatus.InConsultation) ||
+            this.hasUnavailableLinkedParticipants(participant)
+        ) {
             return 'unavailable';
         } else if (participant.status === ParticipantStatus.InConsultation) {
```
[the remark this answers is not in the document]
Comment (phillipjchambers) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts:51]: Not my change, but you're right, it's useless. Removing.

**Review by kevin-hmcts — APPROVED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts [hunk 4]
```diff
@@ -67,4 +75,18 @@ export class IndividualParticipantStatusListComponent extends WRParticipantStatu
...
     }
+
+    private hasUnavailableLinkedParticipants(participant: ParticipantResponse) {
+        if (participant.linked_participants.length) {
+            return participant.linked_participants.some(lp => {
+                const linkedParticipant = this.nonJudgeParticipants.find(p => p.id === lp.linked_id);
+                return (
+                    linkedParticipant &&
+                    linkedParticipant.status !== ParticipantStatus.Available &&
+                    linkedParticipant.status !== ParticipantStatus.InConsultation
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.ts:86]: don't think we need the `InConsultation` here. The participant will either be available or not

**Review by shaed-parkar — APPROVED** [review]
Looks sound though I don't think you need the `InConsultation` status check

**Pull request merged**